### PR TITLE
Port bool-vector, make-bool-vector

### PR DESCRIPTION
**Pull request by nickdrozd**

### CODE REVIEW

File: rust_src/src/alloc.rs [hunk 1]
```diff
@@ -35,4 +36,25 @@ pub fn memory_use_counts() -> Vec<LispObject> {
...
+/// LENGTH must be a number.  INIT matters only in whether it is t or nil.
+#[lisp_fn]
+pub fn make_bool_vector(length: EmacsInt, init: bool) -> LispObject {
+    unsafe { bool_vector_fill(make_uninit_bool_vector(length), init.into()) }
+}
+
+/// Return a new bool-vector with specified arguments as elements.
+/// Any number of arguments, even zero arguments, are allowed.
+/// usage: (bool-vector &rest OBJECTS)
+#[lisp_fn(min = "0")]
```
[the remark this answers is not in the document]
Comment (db48x) [reply rust_src/src/alloc.rs]: It's still required. `eval_sub` checks the min and max arguments before it checks to see if the type is UNEVALLED or MANY (https://github.com/Wilfred/remacs/blob/master/src/eval.c#L1502-L1509).

The `lisp_fn` macro sets the max args to a negative number for these functions, but it doesn't actually help you out with the minimum args; it's an oversight that can be fixed pretty easily.

[the remark this answers is not in the document]
Comment (brotzeit) [reply rust_src/src/alloc.rs]: Oh. There are already similiar functions that don't set the minimum.

[the remark this answers is not in the document]
Comment (db48x) [reply rust_src/src/alloc.rs]: Oops. They probably can't be called with zero arguments then.

**Comment by nickdrozd** [comment]
Maybe there is a masked bug lurking, but as it stands now `min` is not required for `MANY`-valued functions. I took out the `min = "0"` and added a test case to reflect this.

**Comment by nickdrozd** [comment]
You can try executing `(run-hooks)` right now to verify this.

**Comment by shaleh** [comment]
@nickdrozd I thought we had a PR recently which added `min=` to the MANY lisp_fns? Did we miss some or was the PR not completed?

File: rust_src/src/alloc.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (shaleh) [reply rust_src/src/alloc.rs]: The minimum is actually important. It tells people using the function if they have to provide at least one argument. There was a PR recently which updated MANY-style lisp_fns. If any were missed in that PR a small PR to add the minimums is a good idea.

I would rather see the minimum kept than auto generated. It is useful documentation.

**Comment by nickdrozd** [comment]
@shaleh That was for functions like `message` which require at least one argument but can otherwise take any number. Here the issue is functions like `bool-vector` which can take any number of arguments, but don't require any. Compare `(*)` and `(/)`. The former returns `1`, while the latter raises an error. And sure enough, `/` has `min = "1"` and `*` doesn't specify `min`.

`min = "0"` is still required for functions like `forward-char` which can take at most a fixed number of arguments but don't require any.

**Comment by shaleh** [comment]
Ah. Makes sense. Let's make a focused PR that adds min= where needed and move on.

Code review should check this.
The rustize tool supports this.

**Comment by nickdrozd** [comment]
This seems to work fine without `min = "0"`, so I'll merge it without it. If a bug crops up later or we decide it should be there for style reasons, it will be easy enough to add it in.

**Pull request merged**